x-pack/osquerybeat/beater/config_plugin.go

```diff
@@ -190,13 +195,22 @@ func flattenECSMapping(m map[string]interface{}) (ecs.Mapping, error) {
 
 func traverseTree(depth int, ecsm ecs.Mapping, path []string, v interface{}) error {
 	if s, ok := v.(string); ok {
-		if path[len(path)-1] == "field" {
+		if path[len(path)-1] == keyField {
+			if len(path) == 1 {
+				return fmt.Errorf("unexpected top level key '%s': %w", keyField, ErrECSMappingIsInvalid)
+			}
 			ecsm[strings.Join(path[:len(path)-1], ".")] = ecs.MappingInfo{
 				Field: s,
 			}
 		}
 		return nil
-	} else if path[len(path)-1] == "value" {
+	} else if path[len(path)-1] == keyValue {
+		if len(path) == 1 {
+			return fmt.Errorf("unexpected top level key '%s': %w", keyValue, ErrECSMappingIsInvalid)
+		}
+		if len(path) == 1 {
+			return ErrECSMappingIsInvalid
+		}
```

Comment:
nit; Noop branch, we already know `len(path) > 1` when we reach the if statement.

Reply:
yeah, that was updated in the subsequent commit in this PR